### PR TITLE
kubelet: refactor server containerLogs test to table driven test

### PR DESCRIPTION
**Pull request by zhuangqh**
Signed-off-by: zhuangqh <zhuangqhc@gmail.com>

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

refactor kubelet server containerLogs test to table driven test

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/kubelet/server/server_test.go [hunk 1]
```diff
@@ -948,110 +948,47 @@ func setGetContainerLogsFunc(fw *serverTestFramework, t *testing.T, expectedPodN
...
-	body, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		t.Errorf("Error reading container logs: %v", err)
-	}
-	result := string(body)
-	if result != output {
-		t.Errorf("Expected: '%v', got: '%v'", output, result)
-	}
-}
+func int64ptr(i int64) *int64 { return &i }
```
Comment (mattjmcnaughton) [inline pkg/kubelet/server/server_test.go]: Nit: I don't feel super strongly about this, but I feel like this method is kind of overkill.

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm
/ok-to-test

Nice work here! Confirming we retained all the same test coverage?

File: pkg/kubelet/server/server_test.go (hunk 1 above)
Comment (mtaufen) [reply pkg/kubelet/server/server_test.go]: k8s.io/utils/pointer already implements this function, so you can just use that library if you want.

Comment (zhuangqh) [reply pkg/kubelet/server/server_test.go]: updated, thanks! @mattjmcnaughton @mtaufen

**Review by mattjmcnaughton — COMMENTED** [review]
I think you need to update the bazel files: Run `./hack/update-bazel.sh`. Please ping me when the test suite passes and I'll mark lgtm.

**Comment by zhuangqh** [comment]
@mattjmcnaughton updated. thanks for your kindness guide😊

**Review by feiskyer — COMMENTED** [review]
/lgtm

**Review by mattjmcnaughton — APPROVED** [review]
/lgtm

**Pull request merged**